spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/StoredProcExecutor.java

```diff
@@ -153,6 +163,19 @@ public StoredProcExecutor(DataSource dataSource) {
 		Assert.notNull(dataSource, "dataSource must not be null.");
 		this.dataSource = dataSource;
 
+		if (!guavaPresent) {
+			this.jdbcCallOperationsMap = new LinkedHashMap<String, SimpleJdbcCallOperations>() {
```

Comment:
`SuppressWarnings("serial")`

Also, you need to use 

    new LinkedHashMap<String, SimpleJdbcCallOperations>(this.jdbcCallOperationsCacheSize + 1, 0.75f, true) {...}

to enable LRU processing.